### PR TITLE
SQL-956: Add setup instructions to README

### PR DESCRIPTION
**Pull request by varsha804**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,3 +7,16 @@ Connector for BI](https://docs.mongodb.com/bi-connector/current/),
...
+
+## Set up the ODBC driver
+1. If you are not on a Windows machine, install the [unixODBC](http://www.unixodbc.org/) driver manager. Windows users can utilize the built-in driver manager.
+2. Update the values of `Driver`, `Pwd`, `Server`, `User`, and `Database` in `setup/setupDSN.reg`. The value of `Driver` should be the absolute path of `mongo-odbc-driver/target/debug/mongoodbc.dll`.
+3. Open the registry editor and add a new key under `ODBC/ODBCINST.INI` called `ADL_ODBC`. Then, add two sub-entries to `ADL_ODBC`:
+    ````
+    Driver: <path to dll>
+    Setup: <path to dll>
+   ````
+4. Run `reg import "setup/setupDSN.reg"` to populate the registry editor with the correct subkeys, entries, and values. You can run `reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER"` to determine if the command ran successfully.
```
Comment (nbagnard) [inline README.md:20]: The reg query command and using ODBC Admin are both validation steps. Do you want to put them in the same section?

**Review by nbagnard — CHANGES_REQUESTED** [review]
A couple of little things to look into, but the main steps regarding setup script update, setup and validation are good 👍

File: README.md [hunk 2]
```diff
@@ -7,3 +7,16 @@ Connector for BI](https://docs.mongodb.com/bi-connector/current/),
 please see the
 [mongodb/mongo-bi-connector-odbc-driver](https://github.com/mongodb/mongo-bi-connector-odbc-driver)
 repository.
+
+
+## Set up the ODBC driver
+1. If you are not on a Windows machine, install the [unixODBC](http://www.unixodbc.org/) driver manager. Windows users can utilize the built-in driver manager.
+2. Update the values of `Driver`, `Pwd`, `Server`, `User`, and `Database` in `setup/setupDSN.reg`. The value of `Driver` should be the absolute path of `mongo-odbc-driver/target/debug/mongoodbc.dll`.
```
[the remark this answers is not in the document]
Comment (varsha804) [reply README.md]: I wasn't able to set it up on my mac, so I'll just remove this part!

File: README.md [hunk 3]
```diff
@@ -7,3 +7,16 @@ Connector for BI](https://docs.mongodb.com/bi-connector/current/),
...
+3. Open the registry editor and add a new key under `ODBC/ODBCINST.INI` called `ADL_ODBC`. Then, add two sub-entries to `ADL_ODBC`:
```
[the remark this answers is not in the document]
Comment (varsha804) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -7,3 +7,16 @@ Connector for BI](https://docs.mongodb.com/bi-connector/current/),
...
+    ````
+    Driver: <path to dll>
+    Setup: <path to dll>
+   ````
+4. Run `reg import "setup/setupDSN.reg"` to populate the registry editor with the correct subkeys, entries, and values. You can run `reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER"` to determine if the command ran successfully.
+
+There should be a new entry called `ADL_TEST` under `ODBC/ODBC.INI`. If you are on Windows, open the Microsoft ODBC Administrator (64-bit) and verify that "ADL_ODBC_DRIVER" appears under "System DSN".
```
[the remark this answers is not in the document]
Comment (varsha804) [reply README.md]: Done!

File: README.md [hunk 5]
```diff
@@ -9,14 +9,32 @@ please see the
...
     Driver: <path to dll>
     Setup: <path to dll>
    ````
-4. Run `reg import "setup/setupDSN.reg"` to populate the registry editor with the correct subkeys, entries, and values. You can run `reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER"` to determine if the command ran successfully.
 
-There should be a new entry called `ADL_TEST` under `ODBC/ODBC.INI`. If you are on Windows, open the Microsoft ODBC Administrator (64-bit) and verify that "ADL_ODBC_DRIVER" appears under "System DSN".
+Open the Microsoft ODBC Administrator (64-bit) and verify that "ADL_ODBC_DRIVER" appears under "System DSN".
+
+### 32-bit
+Follow the validation steps listed in the 64-bit section, but make sure to use the 32-bit Microsoft ODBC Administrator. Further modifications: the registry keys should be listed under `HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\ODBC` instead of `HKEY_LOCAL_MACHINE\SOFTWARE\ODBC`, and the dll path must point to the 32-bit version of the driver.
```
Comment (nbagnard) [inline README.md:41]: You need to explain that the setup script has to be updated, changing `HKEY_LOCAL_MACHINE\SOFTWARE` to `HKEY_LOCAL_MACHINE\Wow6432Node\SOFTWARE` otherwise, this will just overwrite the existing registry entries.

File: README.md [hunk 6]
```diff
@@ -9,14 +9,32 @@ please see the
 repository.
 
 
-## Set up the ODBC driver
-1. If you are not on a Windows machine, install the [unixODBC](http://www.unixodbc.org/) driver manager. Windows users can utilize the built-in driver manager.
-2. Update the values of `Driver`, `Pwd`, `Server`, `User`, and `Database` in `setup/setupDSN.reg`. The value of `Driver` should be the absolute path of `mongo-odbc-driver/target/debug/mongoodbc.dll`.
-3. Open the registry editor and add a new key under `ODBC/ODBCINST.INI` called `ADL_ODBC`. Then, add two sub-entries to `ADL_ODBC`:
+## Set up the ODBC driver on Windows (architecture-agnostic)
+Note: users can utilize the built-in driver manager.
+1. Update the values of `Driver`, `Pwd`, `Server`, `User`, and `Database` in `setup/setupDSN.reg`. The value of `Driver` should be the absolute path of `mongo-odbc-driver/target/debug/mongoodbc.dll`.
```
Comment (nbagnard) [inline README.md:14]: The lib can be built as debug or release.
Maybe just say :
The absolute path of the `mongoodbc.dll` library. If you just built it, check in your `mongo-odbc-driver/target/` debug or release directories.

**Review by nbagnard — CHANGES_REQUESTED** [review]
Just 2 small comments left

File: README.md (hunk 5 above)
Comment (varsha804) [reply README.md:41]: Added this point to the "set up driver" section

File: Cargo.lock [hunk 7]
```diff
@@ -78,7 +78,7 @@ dependencies = [
  "parking",
```
Comment (nbagnard) [inline Cargo.lock:78]: Did you mean to add this file to the PR? It seems unrelated

File: README.md [hunk 8]
```diff
@@ -7,3 +7,38 @@ Connector for BI](https://docs.mongodb.com/bi-connector/current/),
 please see the
 [mongodb/mongo-bi-connector-odbc-driver](https://github.com/mongodb/mongo-bi-connector-odbc-driver)
 repository.
+
+
+## Set up the ODBC driver on Windows
+Note: users can utilize the built-in driver manager.
+1. Update the values of `Driver`, `Pwd`, `Server`, `User`, and `Database` in `setup/setupDSN.reg`. The value of `Driver` should be the absolute path of `mongoodbc.dll`. This file should be located in either the `mongo-odbc-driver/target/debug` directory or in the release directory.
+
+2. For 32-bit architectures, use the file path `HKEY_LOCAL_MACHINE\Wow6432Node\SOFTWARE` instead of `HKEY_LOCAL_MACHINE\SOFTWARE` in `setupDSN.reg`.
```
Comment (nbagnard) [inline README.md:16]:
```suggestion
2. For 32-bit architectures, update the file path to `HKEY_LOCAL_MACHINE\Wow6432Node\SOFTWARE` instead of `HKEY_LOCAL_MACHINE\SOFTWARE` in `setupDSN.reg`.
```

**Review by nbagnard — APPROVED** [review]
I have 1 suggestion, feel free to update or not, it's not a big deal.
Also, before you merge, can you make sure that we need an update to Cargo.lock as part of this PR? That seems odd.

File: Cargo.lock (hunk 7 above)
Comment (varsha804) [reply Cargo.lock]: Oh weird, I'll remove this file from the diff

**Comment by varsha804** [comment]
evergreen merge
